packages/commonwealth/vite.config.ts

```diff
@@ -0,0 +1,127 @@
+import react from '@vitejs/plugin-react-swc';
+import path from 'path';
+import { Alias, defineConfig } from 'vite';
+import { createHtmlPlugin } from 'vite-plugin-html';
+import tsconfigPaths from 'vite-tsconfig-paths';
+
+const projectRootDir = path.resolve(__dirname);
+
+function createClientResolver(folder: string): Alias {
+  const find = new RegExp(`^${folder}/(.*)$`);
+  return {
+    find,
+    replacement: path.resolve(
+      projectRootDir,
+      'client',
+      'scripts',
+      folder,
+      '$1',
+    ),
+  };
+}
+
+// https://vitejs.dev/config/
+export default defineConfig(({ command }) => {
+  return {
+    plugins: [
+      react(),
+      createHtmlPlugin({
+        entry: `./${command === 'serve' ? 'client/' : ''}scripts/index.tsx`,
+        template: './client/index.html',
+      }),
+      tsconfigPaths(),
+    ],
+    publicDir: 'static',
+    server: {
+      port: 8080,
+      host: 'localhost',
+      proxy: {
+        '/api': {
+          target: 'http://localhost:3000',
```

Comment:
shouldn't host be dynamic, depending on the environment?

Reply:
Good catch, I will make it configurable